x11-base/xlibre-server: add 25.0.0.9, drop 25.0.0.7

callmetango opened the pull request:
- **x11-base/xlibre-server: add 25.0.0.9, drop 25.0.0.7**
- **x11-base/xorg-server: add 25.0.0.9, drop 25.0.0.7**

stefan11111 commented:
> I have added the `xinerama` use flag since it is a meson build option and the compilation with `-Dxinerama=false` has been fixed upstream in https://github.com/X11Libre/xserver/issues/657.

Could you add it to the live ebuild too?

stefan11111 commented:
@callmetango A lot of drivers get installed to /usr/<libdir>/xorg/modules/{drivers,extensions,input}
I think we should still keep them.

Also, whenever we change the directory where modules are installed, users have to `emerge @x11-module-rebuild`. Since this is a new ebuild version, we can add an ewarn that, after this update, they should reemerge the modules.

callmetango commented:
@stefan11111 ping

callmetango commented:
@stefan11111
> While Xlibre does install a pkgconfig file with the expected module dir, not all packages respect it.
> nvidia-drivers, for example, installs it's modules to xorg/modules/{drivers,extensions,input}. That's why the symlinks are needed.

The loader also falls back to loading modules from  `${PREFIX}/lib/xorg/modules`. Please see snippet of `loadmod.c` in https://github.com/X11Libre/ports-gentoo/pull/59#issuecomment-3233331176 .

Quote:

> // now try loading from legacy / unversioned directories

stefan11111 commented:
@callmetango In that case, test building this pr as an update, and then try removing xlibre and it's modules and reinstalling this from scratch, along with modules.

It both work, LGTM.

callmetango commented:
@stefan11111 I removed the `keepdir` stuff from the `xlibre-server-25.0.0.9` ebuild and then tested the whole thing as you requested.

### Test 1

I upgraded from `xlibre-server-25.0.0.8` to `xlibre-server-25.0.0.9` and had a look at the module dirs. They were all fine. I additionally installed `xf86-input-joystick-9999` using one commit before https://github.com/X11Libre/xf86-input-joystick/commit/931dbdbb84eab7ad50ec369d0e46b0c14e65dce6 so it got installed into the non-versioned `/usr/lib/xorg/modules/input` dir. I also added a `module` section file to my `xorg.conf.d` to force the loading of the joystick driver.

Result: It all just worked.

### Test 2

I unmerged the `xlibre-server` ebuild and all installed `xf86-...` drivers and started from scratch including the `xf86-input-joystick` driver and the module configuration from above.

Result: It all just worked.

callmetango commented:
2x yes.
